cudax/include/cuda/experimental/__stf/utility/stackable_ctx.cuh

```diff
@@ -0,0 +1,298 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of CUDASTF in CUDA C++ Core Libraries,
+// under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2022-2024 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+
+/**
+ * @file
+ * @brief Stackable context and logical data to nest contexts
+ */
+
+#pragma once
+
+#include <cuda/__cccl_config>
+
+#if defined(_CCCL_IMPLICIT_SYSTEM_HEADER_GCC)
+#  pragma GCC system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_CLANG)
+#  pragma clang system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_MSVC)
+#  pragma system_header
+#endif // no system header
+
+#include "cuda/experimental/__stf/allocators/adapters.cuh"
+#include "cuda/experimental/stf.cuh"
+
+namespace cuda::experimental::stf {
+
+template <typename T>
+class stackable_logical_data;
+
+class stackable_ctx {
+public:
+    class impl {
+    public:
+        impl() { push(stream_ctx(), nullptr); }
+
+        ~impl() = default;
+
+        void push(context ctx, cudaStream_t stream) {
+            s.push_back(mv(ctx));
+            s_stream.push_back(stream);
+        }
+
+        void pop() {
+            s.back().finalize();
+
+            s.pop_back();
+
+            s_stream.pop_back();
+
+            assert(alloc_adapters.size() > 0);
+            alloc_adapters.back().clear();
+            alloc_adapters.pop_back();
+        }
+
+        size_t depth() const { return s.size() - 1; }
+
+        auto& get() { return s.back(); }
+
+        const auto& get() const { return s.back(); }
+
+        auto& at(size_t level) {
+            assert(level < s.size());
+            return s[level];
+        }
+
+        const auto& at(size_t level) const {
+            assert(level < s.size());
+
+            return s[level];
+        }
```

Comment:
This does not fulfill the contracts of a container `at` function. 

This is rather a `operator[]`

Reply:
i've changed it to []